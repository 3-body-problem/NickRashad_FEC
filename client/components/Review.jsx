import React from 'react';
import * as Icon from './images/Icons.jsx'

const Review = (props) => {
  let { review } = props;
  let rating = `i-stars i-stars--${review.rating * 10}`;
  let date = (new Date(review.createdAt).toLocaleString()).split(',')[0];
  return (
  <div className="review-wrapper yelp-font">
    <div className="review-content">
      <div className={rating}></div><span className="rating-date"> {date} </span>
      { review.checkin === true &&
      <div className="checkin">
        <span className="icon-user-review" style={{width: '18px', height: '18px'}}> <Icon.checkin /> </span>
        <span > 1 check-in</span>
      </div>
      }
      <p className="text">{review.reviewText}</p>
    </div>
    <div className="review-footer">
      <p className="voting-intro">Was this review ...?</p>
      <ul className="voting" >
        <li className="vote-item">
          <span className="icon-user-review" style={{width: '18px', height: '18px'}}> <Icon.useful /> </span>
          <span> Useful</span>
        </li>
        <li className="vote-item">
          <span className="icon-user-review" style={{width: '18px', height: '18px'}}> <Icon.funny /> </span>
          <span> Funny</span>
        </li>
        <li className="vote-item">
          <span className="icon-user-review" style={{width: '18px', height: '18px'}}> <Icon.cool /> </span>
          <span> Cool</span>
        </li>
      </ul>
    </div>
  </div>
)};

export default Review;