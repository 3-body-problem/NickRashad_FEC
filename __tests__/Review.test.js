import React from 'react';
import Review from '../client/components/Review';
import { shallow, mount, render } from 'enzyme';
import { act } from 'react-dom/test-utils';

let review;

describe('Review component', () => {
  beforeAll(() => {
    review = {
      bId: 7,
      uId: 90,
      rating: 1,
      checkin: false,
      createdAt: "2013-02-19T13:16:04.526Z",
      updatedAt: "2013-02-19T13:16:04.526Z",
      reviewText: "My taste buds are still tingling from our last visit! The food was flavorful, savory, and succulent. I want to hire their decorator to furnish my apartment. Overall experience: 5 stars.",
      cool: [{uId: 98, username: "Waylon Reilly"}],
      funny: [{uId: 75, username: "Lavada Hermiston"}, {uId: 7, username: "George Schowalter"}],
      useful: [],
    };
  });

  test('matches snapshot', () => {
    let wrapper = mount(<Review review={review}></Review>);
    // console.log(wrapper.debug());
    expect(wrapper).toMatchSnapshot();
  });

  test('accepts review props', () => {
    let wrapper = mount(<Review review={review}></Review>);
    expect(wrapper.prop('review')).toEqual(review);
  });

  test('test properly renders', () => {
    let wrapper = mount(<Review review={review}></Review>);
    // console.log( wrapper.find('.text').text() );
    expect(wrapper.find('.text').text()).toEqual(review.reviewText);
  });
});