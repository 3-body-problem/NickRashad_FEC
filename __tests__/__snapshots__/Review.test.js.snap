// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Review component matches snapshot 1`] = `
<Review
  review={
    Object {
      "bId": 7,
      "checkin": false,
      "cool": Array [
        Object {
          "uId": 98,
          "username": "Waylon Reilly",
        },
      ],
      "createdAt": "2013-02-19T13:16:04.526Z",
      "funny": Array [
        Object {
          "uId": 75,
          "username": "Lavada Hermiston",
        },
        Object {
          "uId": 7,
          "username": "George Schowalter",
        },
      ],
      "rating": 1,
      "reviewText": "My taste buds are still tingling from our last visit! The food was flavorful, savory, and succulent. I want to hire their decorator to furnish my apartment. Overall experience: 5 stars.",
      "uId": 90,
      "updatedAt": "2013-02-19T13:16:04.526Z",
      "useful": Array [],
    }
  }
>
  <div
    className="review-wrapper yelp-font"
  >
    <div
      className="review-content"
    >
      <div
        className="i-stars i-stars--10"
      />
      <span
        className="rating-date"
      >
         
        2/19/2013
         
      </span>
      <p
        className="text"
      >
        My taste buds are still tingling from our last visit! The food was flavorful, savory, and succulent. I want to hire their decorator to furnish my apartment. Overall experience: 5 stars.
      </p>
    </div>
    <div
      className="review-footer"
    >
      <p
        className="voting-intro"
      >
        Was this review ...?
      </p>
      <ul
        className="voting"
      >
        <li
          className="vote-item"
        >
          <span
            className="icon-user-review"
            style={
              Object {
                "height": "18px",
                "width": "18px",
              }
            }
          >
             
            <useful>
              <svg
                fill="#666"
                height="18"
                width="18"
              >
                <path
                  d="M9 17c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zM9 2C5.14 2 2 5.14 2 9s3.14 7 7 7 7-3.14 7-7-3.14-7-7-7zm2 8.392V12H7v-1.608a3.982 3.982 0 0 1-2-3.445 4 4 0 0 1 8 0c0 1.477-.81 2.752-2 3.445zM8 5.25a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5zm1.003 9.747h-.006A1.997 1.997 0 0 1 7 13h4a1.997 1.997 0 0 1-1.997 1.997z"
                />
              </svg>
            </useful>
             
          </span>
          <span>
             Useful
          </span>
        </li>
        <li
          className="vote-item"
        >
          <span
            className="icon-user-review"
            style={
              Object {
                "height": "18px",
                "width": "18px",
              }
            }
          >
             
            <funny>
              <svg
                fill="#666"
                height="18"
                width="18"
              >
                <path
                  d="M9 17c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zM9 2C5.14 2 2 5.14 2 9s3.14 7 7 7 7-3.14 7-7-3.14-7-7-7zm0 12a4.87 4.87 0 0 1-4.787-4h9.574A4.87 4.87 0 0 1 9 14zm2.5-5.625a1.376 1.376 0 1 1 0-2.75 1.376 1.376 0 0 1 0 2.75zm-5 0a1.376 1.376 0 1 1 0-2.75 1.376 1.376 0 0 1 0 2.75z"
                />
              </svg>
            </funny>
             
          </span>
          <span>
             Funny
          </span>
        </li>
        <li
          className="vote-item"
        >
          <span
            className="icon-user-review"
            style={
              Object {
                "height": "18px",
                "width": "18px",
              }
            }
          >
             
            <cool>
              <svg
                fill="#666"
                height="18"
                width="18"
              >
                <path
                  d="M9 17c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zM9 2C5.14 2 2 5.14 2 9s3.14 7 7 7 7-3.14 7-7-3.14-7-7-7zm2 8.392V12H7v-1.608a3.982 3.982 0 0 1-2-3.445 4 4 0 0 1 8 0c0 1.477-.81 2.752-2 3.445zM8 5.25a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5zm1.003 9.747h-.006A1.997 1.997 0 0 1 7 13h4a1.997 1.997 0 0 1-1.997 1.997z"
                />
              </svg>
            </cool>
             
          </span>
          <span>
             Cool
          </span>
        </li>
      </ul>
    </div>
  </div>
</Review>
`;
